Bugfixes

sfi2022 opened the pull request:
**Description**
Bugfix error message because if dublicate users.  Refactoring id type String to Integer.

Short description or comments
See MPDZ2-1006

**Reference**

Issues #XXX

sfi2022 committed:
Warn at inconsistent active directory group and zammad-ldap-sync
application properties configuration.

sfi2022 committed:
Refactoring id type String to Integer.

sfi2022 committed:
Fixes bugs for users with multiple group_ids. Remove user duplicates and
correct log.error output and aborted follow-up processing.

sfi2022 committed:
Log output adapted to bugfix implementation.

lehju commented on src/main/java/de/muenchen/zammad/ldap/branch/EliminatedLdapUser.java:
```diff
@@ -44,16 +46,16 @@ protected void checkLdapUsers(Map.Entry<String, LdapOuNode> entry, Map<String, E
                         log.error("Inconsistent Zammad state. More than one zammad user found for lhmObjectId '{}' :",
                                 lhmObjectId);
-                        list.forEach(item -> log.error(LOG_ID, item.getId()));
+                        log.error("List without dublicate users: {}." , users.toString());
```
Shouldn't it be “List with duplicate users”?

sfi2022 replied:
Perhaps somewhat misleading. It depends on how you look at it. What is meant is "the previously created list without duplicates is displayed". See https://github.com/it-at-m/zammad-ldap-sync/pull/91/commits/3576be2afe1ddeead5111d7c9cae2aa9668ed740 
If there are still duplicates left, then... :-(  We can make it clearer later.

lehju replied:
All right.

sfi2022 committed:
Update EliminatedLdapUser.java

@lehju ... but you are right. My explanation is a bit out of the box. I think this text explains better why this list is logged.
